test-lib-openshift.sh

```diff
@@ -291,4 +292,74 @@ function ct_os_cluster_running() {
   oc status &>/dev/null
 }
 
+# ct_os_test_s2i_app_func IMAGE APP CONTEXT_DIR CHECK_CMD [OC_ARGS]
+# --------------------
+# Runs [image] and [app] in the openshift and optionally specifies env_params
+# as environment variables to the image. Then check the container by arbitrary
+# function given as argument (such an argument may include <IP> string,
+# that will be replaced with actual IP).
+# Arguments: image - prefix or whole ID of the pod to run the cmd in  (compulsory)
+# Arguments: app - url or local path to git repo with the application sources  (compulsory)
+# Arguments: context_dir - sub-directory inside the repository with the application sources (compulsory)
+# Arguments: check_command - CMD line that checks whether the container works (compulsory; '<IP>' will be replaced with actual IP)
+# Arguments: oc_args - all other arguments are used as additional parameters for the `oc new-app`
+#            command, typically environment variables (optional)
+function ct_os_test_s2i_app_func() {
+  local image_name=${1}
+  local app=${2}
+  local context_dir=${3}
+  local check_command=${4}
+  local oc_args=${5:-}
+  local image_name_no_namespace=${image_name##*/}
+  local service_name="${image_name_no_namespace}-testing"
+  local image_tagged="${image_name_no_namespace}:testing"
+
+  ct_os_new_project
+  # Create a specific imagestream tag for the image so that oc cannot use anything else
+  ct_os_upload_image "${image_name}" "${image_tagged}"
+
+  ct_os_deploy_s2i_image "${image_tagged}" "${app}" \
+                          --context-dir="${context_dir}" \
+                          --name "${service_name}" \
+                          ${oc_args}
+
+  ct_os_wait_pod_ready "${service_name}" 300
+
+  local ip=$(ct_os_get_service_ip "${service_name}")
+  local check_command_exp=$(echo "$command" | sed -e "s/<IP>/$ip/g")
+
+  ${check_command_exp}
```

Comment:
Just a small suggestion, in such situations it is preferred to have quoted command, and:
```patch
- ${check_command_exp}
+ eval "$check_command_exp"
```

Usually we don't have that huge flexibility to be able to put `<IP>` on any position in the
evaluated command.  So one can just run the command + <IP>.  But it is possible to do
```bash
eval set -- "$check_command_exp"
local command=()
for arg; do
  if test "$arg" != "<IP>"; then
    command+=( "$arg" )
  else
    command+=( "$ip" )
  fi
done
"${command[@]}"
```

But that's not an issue;  I wouldn't be against merging this "as-is"; that's a new so please don't feel blocked and (from my POV) feel free to just push to master :-).

Reply:
@praiskup I'll add the `eval` as suggested, although I did not understand the point about `<IP>` -- with the proposed changes, it looks like it wouldn't be possible to use `<IP>` as part of the string, like in a very simple scenario one might provide a check command just like this:
```
curl http://<IP>/api/check | grep true
```

Reply:
Right, I misunderstood the semantics.  Thanks.